src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1358 @@
+
+    def forward(self, hidden_states, split_position=-1):
+        if split_position == -1:
+            return self.text(hidden_states)
+        if split_position == 0:
+            return self.image(hidden_states)
+        text_hidden, image_hidden = torch.split(
+            hidden_states,
+            [split_position, hidden_states.size(1) - split_position],
+            dim=1,
+        )
+        text_hidden, image_hidden = self.text(text_hidden), self.image(image_hidden)
+        return torch.cat([text_hidden, image_hidden], dim=1)
+
+
+class Beit3LayerNorm(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.text = nn.LayerNorm(config.embed_dim, eps=config.layer_norm_eps)
+        self.image = nn.LayerNorm(config.embed_dim, eps=config.layer_norm_eps)
+
+    def forward(self, hidden_states, split_position=-1):
+        if split_position == -1:
+            return self.text(hidden_states)
+
+        if split_position == 0:
+            return self.image(hidden_states)
+
+        text_hidden, image_hidden = torch.split(
+            hidden_states,
+            [split_position, hidden_states.size(1) - split_position],
+            dim=1,
+        )
+        text_hidden = self.text(text_hidden)
+        image_hidden = self.image(image_hidden)
+        hidden_states = torch.cat([text_hidden, image_hidden], dim=1)
+        return hidden_states
+
+
+class Beit3Embedder(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        img_size = (config.img_size, config.img_size)
+        patch_size = (config.patch_size, config.patch_size)
+        self.num_patches = (img_size[1] // patch_size[1]) * (img_size[0] // patch_size[0]) + 1
+
+        self.text = nn.Embedding(self.num_patches + 2, config.embed_dim)
+        self.image = nn.Embedding(config.max_source_positions, config.embed_dim)
+
+    def forward(self, hidden_states, text_end_position, multiway_split_position=-1):
+        if text_end_position is None:
+            positions = torch.arange(2, hidden_states.size(1) + 2, device=hidden_states.device).long().unsqueeze(0)
+        else:
+            positions = text_end_position
+        if multiway_split_position == -1:
+            return self.text(positions)
+        if multiway_split_position == 0:
+            return self.image(positions)
+        text_hidden, image_hidden = torch.split(
+            hidden_states,
+            [multiway_split_position, hidden_states.size(1) - multiway_split_position],
+            dim=1,
+        )
+        if text_end_position is None:
+            text_positions = torch.arange(2, text_hidden.size(1) + 2, device=text_hidden.device).long().unsqueeze(0)
+        else:
+            text_positions = text_end_position
+        if text_end_position is None:
+            image_positions = torch.arange(2, image_hidden.size(1) + 2, device=image_hidden.device).long().unsqueeze(0)
+        else:
+            image_positions = text_end_position
```

Comment:
```suggestion
        if text_end_position is None:
            text_positions = torch.arange(2, text_hidden.size(1) + 2, device=text_hidden.device).long().unsqueeze(0)
            image_positions = torch.arange(2, image_hidden.size(1) + 2, device=image_hidden.device).long().unsqueeze(0)
        else:
            text_positions = text_end_position
            image_positions = text_end_position
```
seems to be checking the same thing here no?